gobblin-metastore/src/main/java/org/apache/gobblin/MysqlDagStore.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin;
+
+import java.io.IOException;
+
+import javax.sql.DataSource;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.gobblin.configuration.State;
+import org.apache.gobblin.metastore.MysqlStateStore;
+import org.apache.gobblin.service.ServiceConfigKeys;
+
+
+@Slf4j
+/**
+ * An implementation of {@link MysqlStateStore} backed by MySQL to store Dag.
+ *
+ * @param <T> state object type
+ **/
+public class MysqlDagStore<T extends State> extends MysqlStateStore<T> {
```

Comment:
Do we need a separate MysqlDagStore, or we can just bump the lengths in the base class?

Reply:
The base class MysqlStateStore is a base class and is shared by a lot of implementations. I found it cleaner this way.

Reply:
So is this length problem only present for dag store? I would expect that if the flow name is part of the key, it will affect other state stores as well. I see that we have at least a "job_status" state store in shared Gobblin service. And if the new state store is created, it will also hit this problem.

Reply:
Yes, we have job_status store and flow spec store besides dag store.
job_status store is also a MysqlStateStore like this dag store. Spec store is different in its own.
So I have created a new method getCreateJobStateTableTemplate that new stores should override to enforce their own limits.

Reply:
Is there ever a scenario where we want to use the old dag state store?

Reply:
Both the dag stores are actually the same. New store extends the old store overriding one method here https://github.com/apache/gobblin/pull/3298/files/d45e2c83eab8b6420fc90162ef27f1e3f8d0a9c9#diff-e44b7b6bb150dd599827d36ca60c5a08e31e406013cfb22d3b363f794899f4e8R52